### PR TITLE
ExtParamConfig and GenKwConfig refactor

### LINKED ISSUE #5345: Make GenKwConfig more similar to FieldConfig
When refactoring the various enkf_config_node(s), certain nodes where re-implemented slightly different.
Using the field config implementation as a reference, see if we can improve GenKwConfig some.

### PR DESCRIPTION
**Pull request by DanSava**
**Issue**
Resolves #5345 

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/analysis/_es_update.py [hunk 1]
```diff
@@ -191,25 +191,16 @@ def _save_temporary_storage_to_disk(
...
-                key,
-                parameter_keys,
-                iens_active_index,
-                matrix,
-            )
-        elif isinstance(config_node, SurfaceConfig):
-            for i, realization in enumerate(iens_active_index):
+        for i, realization in enumerate(iens_active_index):
+            if isinstance(config_node, GenKwConfig):
+                parameter_keys = list(ensemble_config.get_keyword_model_config(key))
```
Comment (andreas-el) [inline src/ert/analysis/_es_update.py:196]: Does this make sense? Isn't config_node already `ensemble_config.get_keyword_model_config(key)` ?

```
    def get_keyword_model_config(self, key: str) -> GenKwConfig:
        return self[key]
```

File: src/ert/_c_wrappers/enkf/ensemble_config.py [hunk 2]
```diff
@@ -404,13 +425,7 @@ def hasNodeGenData(self, key: str) -> bool:
...
     def getNode(
         self, key: str
-    ) -> Union[
-        ParameterConfig,
-        GenKwConfig,
-        GenDataConfig,
-        SummaryConfig,
-        ExtParamConfig,
-    ]:
+    ) -> Union[ParameterConfig, GenDataConfig, SummaryConfig, ExtParamConfig,]:
```
Comment (andreas-el) [inline src/ert/_c_wrappers/enkf/ensemble_config.py:428]: Can this be further simplified by removing `ExtParamConfig`?

File: src/ert/analysis/_es_update.py (hunk 1 above)
Comment (DanSava) [reply src/ert/analysis/_es_update.py:196]: Good point, this should be equivalent to:

```python

if isinstance(config_node, GenKwConfig):
                parameter_keys = list(config_node)
```

File: src/ert/_c_wrappers/enkf/ensemble_config.py (hunk 2 above)
Comment (DanSava) [reply src/ert/_c_wrappers/enkf/ensemble_config.py:428]: yes

**Review by andreas-el — APPROVED** [review]
Well done! Nice! :rocket:

**Pull request merged**